packages/contracts-bedrock/scripts/L2Genesis.s.sol

```diff
@@ -0,0 +1,373 @@
+    function run() public {
+        _setPrecompiles();
+        _setProxies();
+        _setImplementations();
+
+        vm.dumpState(outfile);
+
+        for (uint256 i = 0; i < addrToRemoveFromAlloc.length; i++) {
+            _removeAllocFromGenesisFile(outfile, addrToRemoveFromAlloc[i]);
+        }
+    }
+
+    /// @notice Give all of the precompiles 1 wei so that they are
+    ///         not considered empty accounts.
+    function _setPrecompiles() internal {
+        for (uint256 i; i < PRECOMPILE_COUNT; i++) {
+            vm.deal(address(uint160(i)), 1);
+        }
+    }
+
+    /// @dev Set up the accounts that correspond to the predeploys.
+    ///      The Proxy bytecode should be set. All proxied predeploys should have
+    ///      the 1967 admin slot set to the ProxyAdmin predeploy. All defined predeploys
+    ///      should have their implementations set.
+    function _setProxies() internal {
+        bytes memory code = vm.getDeployedCode("Proxy.sol:Proxy");
+        uint160 prefix = uint160(0x420) << 148;
+        for (uint256 i = 0; i < PROXY_COUNT; i++) {
+            address addr = address(prefix | uint160(i));
+            if (_notProxied(addr)) {
+                continue;
+            }
+
+            vm.etch(addr, code);
+            vm.store(addr, PROXY_ADMIN_ADDRESS, bytes32(uint256(uint160(Predeploys.PROXY_ADMIN))));
+
+            if (_isDefinedPredeploy(addr)) {
+                address implementation = _predeployToCodeNamespace(addr);
+                vm.store(addr, PROXY_IMPLEMENTATION_ADDRESS, bytes32(uint256(uint160(implementation))));
+            }
+        }
+    }
+
+    /// @notice LEGACY_ERC20_ETH is not being predeployed
+    function _setImplementations() internal {
+        _setLegacyMessagePasser();
+        _setDeployerWhitelist();
+        _setWETH9();
+        _setL2StandardBridge();
+        _setL2CrossDomainMessenger();
+        _setSequencerFeeVault();
+        _setOptimismMintableERC20Factory();
+        _setL1BlockNumber();
+        _setGasPriceOracle();
+        _setGovernanceToken();
+        _setL1Block();
+    }
+
+    function _setLegacyMessagePasser() internal {
+        address impl = _predeployToCodeNamespace(Predeploys.LEGACY_MESSAGE_PASSER);
+        vm.etch(
+            impl,
+            vm.getDeployedCode("LegacyMessagePasser.sol:LegacyMessagePasser")
+        );
+
+        _verifyProxyImplementationAddress(Predeploys.LEGACY_MESSAGE_PASSER, impl);
+    }
+
+    function _setDeployerWhitelist() internal {
+        address impl = _predeployToCodeNamespace(Predeploys.DEPLOYER_WHITELIST);
+        vm.etch(
+            _predeployToCodeNamespace(impl),
+            vm.getDeployedCode("DeployerWhitelist.sol:DeployerWhitelist")
+        );
+
+        _verifyProxyImplementationAddress(Predeploys.DEPLOYER_WHITELIST, impl);
+    }
+
+    /// @notice Contract metadata appended to deployed bytecode will differ
+    ///         from previous L2 genesis output. Additionally, this contract
```

Comment:
> Contract metadata appended to deployed bytecode will differ from previous L2 genesis output

What does this mean?

Reply:
Maybe there's a better way to word this, but I'm talking about the compiler metadata that gets appended to the bytecode is different from current L2 genesis (using the Go code), otherwise the bytecode is the same

![image](https://github.com/ethereum-optimism/optimism/assets/20375610/63a57999-be85-481d-af54-bc850547247e)

Reply:
Ah I see. I think if you just edit the comment to say the "metadata hash" instead of just metadata, and note that it differs because (presumably) we re-compiled the WETH9 source, that would explain it

Reply:
Updated!